src/pkg/utils/utils_cgo.go

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright © 2019 – 2022 Red Hat Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package utils
+
+import (
+	"errors"
+	"fmt"
+	"os/user"
+	"unsafe"
+)
+
+/*
+#cgo LDFLAGS: -ldl
+
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#include <dlfcn.h>
+
+#include <shadow/subid.h>
+
+#if SUBID_ABI_MAJOR < 4
+# define subid_get_uid_ranges get_subuid_ranges
+# define subid_get_gid_ranges get_subgid_ranges
+#endif
+
+#define TOOLBOX_STRINGIFY_HELPER(s) #s
+#define TOOLBOX_STRINGIFY(s) TOOLBOX_STRINGIFY_HELPER (s)
+
+typedef bool (*ToolboxSubidInitFunc) (const char *progname, FILE *logfd);
+typedef int (*ToolboxSubidGetRangesFunc) (const char *owner, struct subid_range **ranges);
+
+const char *TOOLBOX_LIBSUBID = "libsubid.so." TOOLBOX_STRINGIFY (SUBID_ABI_VERSION);
+
+const char *TOOLBOX_LIBSUBID_INIT = "libsubid_init";
+const char *TOOLBOX_SUBID_INIT = "subid_init";
+
+const char *TOOLBOX_SUBID_GET_UID_RANGES_SYMBOL = TOOLBOX_STRINGIFY (subid_get_uid_ranges);
+const char *TOOLBOX_SUBID_GET_GID_RANGES_SYMBOL = TOOLBOX_STRINGIFY (subid_get_gid_ranges);
+
+void
+toolbox_subid_init (void *subid_init_func)
+{
+  (* (ToolboxSubidInitFunc) subid_init_func) (NULL, NULL);
```

Comment:
It's this file descriptor that's used by `libsubid.so` for logging that's getting leaked.  There's no `subid_close` counterpart to `subid_init`, so I don't think we can do much here.

Reply:
Wait a minute!  We can provide a `FILE *` that's under our control, instead of passing a `NULL` and letting `libsubid.so` create the `FILE *` that eventually gets leaked.

I suppose the easiest option is to pass `stderr`.  It's the least amount of code, and it doesn't look like `libsubid.so` is doing any logging in the specific code paths that we are using.  If this proves to be noisy, then we could check the debug level and conditionally pass `stderr` or a `FILE *` pointing to `/dev/null`.

Reply:
This seems to have done the trick.